plugins/modules/ec2_vpc_route_table.py

```diff
@@ -658,28 +742,26 @@ def ensure_route_table_present(connection, module):
 
     if routes is not None:
         result = ensure_routes(connection=connection, module=module, route_table=route_table,
-                               route_specs=routes, propagating_vgw_ids=propagating_vgw_ids,
-                               check_mode=module.check_mode, purge_routes=purge_routes)
-        changed = changed or result['changed']
+                               route_specs=routes, purge_routes=purge_routes)
+        changed = changed or result
 
     if propagating_vgw_ids is not None:
         result = ensure_propagation(connection=connection, module=module, route_table=route_table,
-                                    propagating_vgw_ids=propagating_vgw_ids, check_mode=module.check_mode)
-        changed = changed or result['changed']
+                                    propagating_vgw_ids=propagating_vgw_ids)
+        changed = changed or result
 
     if not tags_valid and tags is not None:
         changed |= ensure_ec2_tags(connection, module, route_table['RouteTableId'],
                                    tags=tags, purge_tags=purge_tags,
                                    retry_codes=['InvalidRouteTableID.NotFound'])
         route_table['Tags'] = describe_ec2_tags(connection, module, route_table['RouteTableId'])
 
-    if subnets is not None:
-        associated_subnets = find_subnets(connection, module, vpc_id, subnets)
-
-        result = ensure_subnet_associations(connection=connection, module=module, route_table=route_table,
-                                            subnets=associated_subnets, check_mode=module.check_mode,
-                                            purge_subnets=purge_subnets)
-        changed = changed or result['changed']
+    associated_subnets = find_subnets(connection, module, vpc_id, subnets)
+    subnets_result = ensure_subnet_associations(connection=connection, module=module, route_table=route_table,
+                                                subnets=associated_subnets, purge_subnets=purge_subnets)
+    gateway_result = ensure_gateway_association(connection=connection, module=module, route_table=route_table,
+                                                gateway_id=gateway_id)
+    changed = changed or subnets_result or gateway_result
```

Comment:
Why not keeping as it is and doing something similar for the gateway_id? 
```suggestion
    if subnets:
       associated_subnets = find_subnets(connection, module, vpc_id, subnets)
       subnets_result = ensure_subnet_associations(connection=connection, module=module, route_table=route_table, subnets=associated_subnets, purge_subnets=purge_subnets)
       changed = changed or result['changed']
    if gateway_id:
       gateway_result = ensure_gateway_association(connection=connection, module=module, route_table=route_table, gateway_id=gateway_id)
       changed = changed or gateway_result
```

Reply:
I originally did but I changed it to be able to dissociate a gw when none is passed in. If I add a purge_gw param id revert this